akka-javasdk-testkit/src/test/java/akka/javasdk/testkit/TestModelProviderTest.java

```diff
@@ -130,9 +137,56 @@ void testMockResponseWithPredicate() {
     assertThat(goodbyeResponse.aiMessage().text()).isEqualTo("Goodbye response");
   }
 
+  @Test
+  void testFailWhenActionIsMissing() {
+
+    var whenClause =
+    testModelProvider
+      .whenMessage("hello");
+
+    ChatModel chatModel = (ChatModel) testModelProvider.createChatModel();
+    ChatRequest helloRequest = ChatRequest.builder()
+      .messages(List.of(new UserMessage("hello")))
+      .build();
+
+    try {
+      chatModel.doChat(helloRequest);
+      fail("Should have failed");
+    } catch (RuntimeException e) {
+      assertThat(e.getMessage()).startsWith("A matching predicate was defined for");
+      return;
+    }
+
+    // adding a reply replaces the catcher for MissingModelResponseException
+    whenClause.reply("Hello response");
+    ChatResponse helloResponse = chatModel.doChat(helloRequest);
+    assertThat(helloResponse.aiMessage().text()).isEqualTo("Hello response");
+  }
+
+  @Test
+  void testFailure() {
+
+    testModelProvider.whenMessage("hello").failWith(new RuntimeException("I can't handle this"));
+
+    ChatModel chatModel = (ChatModel) testModelProvider.createChatModel();
+    ChatRequest helloRequest = ChatRequest.builder()
+      .messages(List.of(new UserMessage("hello")))
+      .build();
+    try {
+      chatModel.doChat(helloRequest);
+      fail("Should have failed");
+    } catch (RuntimeException e) {
+      assertThat(e.getMessage()).isEqualTo("I can't handle this");
```

Comment:
This kind of assertion is only possible for us. Here we are testing that we can configure the model provider with a failure. 

Final users won't have access to the exception message. They get a generic "ComponentClient error" one.